dask_ml/model_selection/_incremental.py

```diff
@@ -347,3 +354,228 @@ def fit(
         scorer=scorer,
         random_state=random_state,
     )
+
+
+# ----------------------------------------------------------------------------
+# Base class for scikit-learn compatible estimators using fit
+# ----------------------------------------------------------------------------
+
+
+class BaseIncrementalSearchCV(DaskBaseSearchCV):
+    """Base class for estimators using the incremental `fit`.
+
+    Subclasses must implement the following abstract methods
+
+    * _get_ids_and_args : Build the dict of
+       ``{model_id : (param_list, additional_calls)}``
+        that are eventually passed to `fit`
```

Comment:
I would expect the `additional_calls` function to be a method of this class that developers would override.

I think that we probably don't need users to define `model_id`s.  The `fit` function will do this for us.  

I do think that we need a way to decide initial parameter lists.  This might be an overrideable method.

Reply:
@mrocklin does `_get_ids_and_args` make sense now? Initially the building of `param_lists` and `additional_calls` was two separate methods, but it seems like they may be dependent on eachother. So I figured having a single method to override the things passed into `fit` made the most sense.

This is the only mandatory method that has to be implemented by subclasses now.